### PR TITLE
 add platform specific includes in diligent-core/2.5.2

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-core/2.5.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
blocks https://github.com/conan-io/conan-center-index/pull/10732

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @jgsogo @SpaceIm @madebr 
please approve

**Review by SSE4 — APPROVED** [review]

File: recipes/diligent-core/all/conanfile.py [hunk 1]
```diff
@@ -184,8 +184,13 @@ def package_info(self):
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Common", "interface"))
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Primitives", "interface"))
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "interface"))
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Android", "interface"))
```
Comment (madebr) [inline recipes/diligent-core/all/conanfile.py:187]: I suppose we're relying on conan here to prune the non-existing include paths?

Comment (AndreyMlashkin) [reply recipes/diligent-core/all/conanfile.py:187]: As far as I know, there is no problem having a non-existing path in PATH variable, while using Windows, Linux or Mac. I can't say anything about mobile OS

**Review by madebr — DISMISSED** [review]

File: recipes/diligent-core/all/conanfile.py [hunk 2]
```diff
@@ -184,8 +184,13 @@ def package_info(self):
...
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Apple", "interface"))
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Basic", "interface"))
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Emscripten", "interface"))
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Linux", "interface"))
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "UWP", "interface"))
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Win32", "interface"))
```
Comment (SpaceIm) [inline recipes/diligent-core/all/conanfile.py:193]: Do not close discussions so quickly please (https://github.com/conan-io/conan-center-index/pull/10780#discussion_r872308956).
AFAIK, non-existing includedirs can break downstream project relying on `CMakeDeps` generator.

Comment (SSE4) [reply recipes/diligent-core/all/conanfile.py:193]: you may keep them alive by creating an empty file

Comment (AndreyMlashkin) [reply recipes/diligent-core/all/conanfile.py:193]: I have closed it as @madebr has approved a MR, meaning he agrees with what I have written.

File: recipes/diligent-core/all/conanfile.py [hunk 3]
```diff
@@ -182,20 +182,30 @@ def package_info(self):
...
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Graphics", "GraphicsAccessories", "interface"))
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Graphics", "GraphicsTools", "interface"))
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Graphics", "HLSL2GLSLConverterLib", "interface"))
         self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Graphics", "Archiver", "interface"))
 
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Primitives", "interface"))
+        self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Basic", "interface"))
+        if self.settings.os == "Android":
+            self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Android", "interface"))
+        elif self.settings.os == "Macos" or self.settings.os == "iOS":
```
Comment (SpaceIm) [inline recipes/diligent-core/all/conanfile.py:198]:
```suggestion
        elif tools.is_apple_os(self.settings.os):
```

File: recipes/diligent-core/all/conanfile.py [hunk 4]
```diff
@@ -182,20 +182,30 @@ def package_info(self):
...
+            self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Apple", "interface"))
+        elif self.settings.os == "Emscripten":
+            self.cpp_info.includedirs.append(os.path.join("include", "DiligentCore", "Platforms", "Emscripten", "interface"))
+        elif self.settings.os == "Linux":
```
Comment (SpaceIm) [inline recipes/diligent-core/all/conanfile.py:202]: FreeBSD also?

Comment (SSE4) [reply recipes/diligent-core/all/conanfile.py:202]: this seems to be linux-specific

**Review by SSE4 — DISMISSED** [review]

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @jgsogo @SpaceIm @madebr @lasote @prince-chrismc  @toge 
please approve

**Review by SSE4 — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Pull request merged**